app/src/common/shared/org/mozilla/vrbrowser/ui/NavigationURLBar.java

```diff
@@ -111,6 +111,8 @@ public void setURL(String aURL) {
                 return;
             else if (aURL.startsWith("resource:") || SessionStore.get().isHomeUri(aURL))
                 aURL = "";
+            else if (aURL.startsWith("data:") && SessionStore.get().isCurrentSessionPrivate())
```

Comment:
what if I want to load my own, a different, data URI in Private Browsing mode? I do this often on desktop (e.g., `data:text/html,&times;`). admittedly, I wouldn't expect this to be done by a non-developer, but this will certainly cause a minor bug.

Reply:
Good call, we could add some logic to more properly handle the private mode page data URL masking without affecting other data URLs.